app/src/test/java/ingsw/group1/findmyphone/location/GeoPositionTest.java

```diff
@@ -0,0 +1,176 @@
+        assertEquals(expectedLongitude, examplePosition.getLongitude());
+    }
+
+    /**
+     * Testing {@link GeoPosition#equals(Object)} returns false for a non {@link GeoPosition}
+     * Object.
+     */
+    @Test
+    public void equals_falseForNonPosition() {
+        assertNotEquals(examplePosition, "Hello there");
+    }
+
+    /**
+     * Testing {@link GeoPosition#equals(Object)} returns false for a {@link GeoPosition} Object
+     * that does not match the equation criteria.
+     */
+    @Test
+    public void equals_falseForNonEqualPosition() {
+        //Latitude and longitude are switched.
+        GeoPosition notTheSamePosition = new GeoPosition(
+                exampleLongitude,
+                exampleLatitude
+        );
+        assertNotEquals(notTheSamePosition, examplePosition);
+    }
+
+    /**
+     * Testing {@link GeoPosition#equals(Object)} returns true for a {@link GeoPosition} Object
+     * that does match the equation criteria.
+     */
+    @Test
+    public void equals_trueForEqualPosition() {
+        GeoPosition theSamePosition = new GeoPosition(
+                exampleLatitude,
+                exampleLongitude
+        );
+        assertEquals(examplePosition, theSamePosition);
+    }
+
+    /**
+     * Testing that {@link GeoPosition#toString()} returns a {@link String} formatted as stated
+     * in the specs:
+     * [latitude]{@link GeoPosition#POSITION_SPLIT_SEQUENCE}[longitude]
+     */
+    @Test
+    public void toString_actsAsStated() {
+        String splitSequence = GeoPosition.POSITION_SPLIT_SEQUENCE;
+        String expectedString = exampleLatitude + splitSequence + exampleLongitude;
+        assertEquals(expectedString, examplePosition.toString());
+    }
+
+    /**
+     * Testing that {@link GeoPosition#GeoPosition(String)} does not accept a String that splits
+     * into more than two parts.
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void stringConstructor_failsForInvalidSplitString() {
+        String splitSequence = GeoPosition.POSITION_SPLIT_SEQUENCE;
+        String invalidString = "a" + splitSequence + "b" + splitSequence + "c";
+        new GeoPosition(invalidString);
+    }
+
+    /**
+     * Testing that {@link GeoPosition#GeoPosition(String)} does not accept a String that splits
+     * into two parts that are not parsable into {@link Double} values.
+     */
+    @Test(expected = NumberFormatException.class)
+    public void stringConstructor_failsForNonDoubleStrings() {
+        String splitSequence = GeoPosition.POSITION_SPLIT_SEQUENCE;
+        String invalidString = "General Kenobi" + splitSequence + "You are a bold one";
+        new GeoPosition(invalidString);
+    }
+
+    /**
+     * Testing that {@link GeoPosition#GeoPosition(String)} accepts the result of
+     * {@link GeoPosition#toString()} as its parameter.
+     */
+    @Test
+    public void stringConstructor_acceptsToStringResult() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'stringConstructor_acceptsToStringResult' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4950325)